src/btree/bt_vrfy.c

```diff
@@ -280,12 +284,36 @@ __verify_key_hs(WT_SESSION_IMPL *session, WT_ITEM *key, WT_VSTUFF *vs)
 #else
         WT_UNUSED(vs);
 #endif
-    }
 
+        /* Perform Verification of data continuity. */
+        if (prev_start.timestamp < stop.timestamp) {
+            WT_ERR_MSG(session, WT_ERROR,
+              "In the Btree %" PRIu32
+              ", Key %s has a overlap of "
+              "timestamp ranges between history store stop timestamp %s being "
+              "newer than a more recent timestamp range having start timestamp %s",
+              hs_btree_id, __wt_buf_set_printable(session, hs_key->data, hs_key->size, vs->tmp1),
+              __verify_timestamp_to_pretty_string(stop.timestamp),
+              __verify_timestamp_to_pretty_string(prev_start.timestamp));
+        }
+        if (prev_start.txnid < stop.txnid) {
+            WT_ERR_MSG(session, WT_ERROR,
+              "In the Btree %" PRIu32
+              ", Key %s has a overlap of "
+              "timestamp ranges between history store stop transaction (%" PRIu64
+              ") being "
+              "newer than a more recent timestamp range having start transaction (%" PRIu64 ")",
+              hs_btree_id, __wt_buf_set_printable(session, hs_key->data, hs_key->size, vs->tmp1),
+              stop.txnid, prev_start.txnid);
+        }
+        prev_start.timestamp = start.timestamp;
+        prev_start.txnid = start.txnid;
+    }
 err:
     __wt_scr_free(session, &hs_key);
     WT_RET(__wt_hs_cursor_close(session, &hs_cursor, session_flags));
 
+    WT_RET_NOTFOUND_OK(ret);
```

Comment:
Sulabh encountered an issue here, so the merge conflict you have now is a fix for that.